cheddar/cheddar-integration-aws/src/main/java/com/clicktravel/infrastructure/persistence/aws/s3/S3FileStore.java

```diff
@@ -75,33 +75,41 @@ public FileItem read(final FilePath filePath) throws NonExistentItemException {
         checkInitialization();
         final GetObjectRequest getObjectRequest = new GetObjectRequest(bucketNameForFilePath(filePath),
                 filePath.filename());
-        try {
-            final S3Object s3Object = amazonS3Client.getObject(getObjectRequest);
+        FileItem fileItem = null;
+        try (final S3Object s3Object = amazonS3Client.getObject(getObjectRequest)) {
+
             final Map<String, String> userMetaData = getUserMetaData(s3Object);
             final String filename = userMetaData.get(USER_METADATA_FILENAME);
-            final String lastUpdatedTimeStr = userMetaData.get(USER_METADATA_LAST_UPDATED_TIME);
-            DateTime lastUpdatedTime = null;
-            if (lastUpdatedTimeStr != null) {
-                try {
-                    lastUpdatedTime = formatter.parseDateTime(lastUpdatedTimeStr);
-                } catch (final Exception e) {
-                    logger.warn(e.getMessage(), e);
-                }
-            }
+            final DateTime lastUpdatedTime = getLastUpdatedTime(userMetaData);
             try {
 
-                final FileItem fileItem = new FileItem(filename, s3Object.getObjectContent(), lastUpdatedTime);
-                return fileItem;
+                fileItem = new FileItem(filename, s3Object.getObjectContent(), lastUpdatedTime);
             } catch (final IOException e) {
                 throw new IllegalStateException(e);
             }
         } catch (final AmazonS3Exception e) {
             if (missingItemErrorCodes.contains(e.getErrorCode())) {
-                throw new NonExistentItemException("Item does not exist" + filePath.directory() + "->"
-                        + filePath.filename());
+                throw new NonExistentItemException(
+                        "Item does not exist" + filePath.directory() + "->" + filePath.filename());
             }
             throw e;
+        } catch (final IOException e1) {
+            logger.error("Error occured while closing S3Object Stream");
```

Comment:
ERROR level is probably too severe, WARN would be more appropriate. Also, it would be more informative if the root cause of the problem were included. Putting these together, we have:
```java
logger.warn("Error closing S3Object", e1);
```